crates/ruff_linter/src/rules/ruff/rules/unnecessary_iterable_allocation_for_first_element.rs

```diff
@@ -209,7 +214,24 @@ fn match_iteration_target(expr: &Expr, semantic: &SemanticModel) -> Option<Itera
                 iterable: true,
             },
         },
+        Expr::List(ast::ExprList { elts, .. }) => {
+            let [elt] = elts.as_slice() else {
+                return None;
+            };
+            let Expr::Starred(ast::ExprStarred { value, .. }) = elt else {
+                return None;
+            };
 
+            let iterable = if value.is_call_expr() {
+                is_func_builtin_iterator(&value.as_call_expr()?.func, semantic)
+            } else {
+                false
+            };
+            IterationTarget {
+                range: value.range(),
```

Comment:
I think we can just use the range of the value here. (The previous logic would have failed if there was, e.g., space between the bracket and the star.)

Reply:
I did not think of that... 

I was inspired by the logic done for the `Expr::ListComp` above.